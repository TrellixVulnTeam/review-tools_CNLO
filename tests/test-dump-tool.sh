#!/bin/sh
set -e

help() {
    cat <<EOM
$ $(basename "$0")
EOM
}

if [ "$1" = "help" ] || [ "$1" = "--help" ] || [ "$1" = "-h" ]; then
    help
    exit
fi

tmpdir=$(mktemp -d)
tmp="$tmpdir/out"
orig_dir=$(pwd)

run() {
    echo "Running: dump-tool $*" | tee -a "$tmp"
    run_no_header "$@" 2>&1 | tee -a "$tmp"
    echo "" | tee -a "$tmp"
}

run_no_header() {
    PYTHONPATH="$orig_dir" "$orig_dir"/bin/dump-tool "$@"
}

comment() {
    echo "$@" | tee -a "$tmp"
}

# --file
comment "= Test --file ="
run --file "./tests/test-store-dump.1"
run --file "./tests/test-store-dump.2"
run_no_header --file "./tests/test-store-dump.1" --output=stdout > "$tmpdir/tst.file.stdout"
diff "./tests/test-store-dump.1.json" "$tmpdir/tst.file.stdout"
run_no_header --file "./tests/test-store-dump.1" --output="$tmpdir/tst.file.out"
diff "./tests/test-store-dump.1.json" "$tmpdir/tst.file.out"

# --file (bad)
comment "= Test --file (bad) ="
cd "$tmpdir"
echo bad > "./bad"
run --file "./bad" --output=stdout
cd "$orig_dir"

# --file --progress
comment "= Test --file --progress ="
run --progress --file "./tests/test-store-dump.1"

# --file --warn-if-empty
comment "= Test --file --warn-if-empty ="
run --warn-if-empty --file "./tests/test-store-dump.1"

# --db-file
comment "= Test --db-file ="
run --db-file "./tests/test-store-dump.1.json"
run --db-file "./tests/test-store-dump.2.json"
run_no_header --db-file "./tests/test-store-dump.1.json" --output=stdout > "$tmpdir/tst.db-file.stdout"
diff "./tests/test-store-dump.1.json" "$tmpdir/tst.db-file.stdout"
run_no_header --db-file "./tests/test-store-dump.1.json" --output="$tmpdir/tst.db-file.out"
diff "./tests/test-store-dump.1.json" "$tmpdir/tst.db-file.out"

# --db-file (bad)
comment "= Test --db-file (bad) ="
cd "$tmpdir"
echo bad > "./bad"
run --db-file "./bad" --output=stdout
cd "$orig_dir"

# --file --db-file (merge)
comment "= Test --file --db-file (merge) ="
run --force-merge --db-file "./tests/test-store-dump.1.json" --file "./tests/test-store-dump.2"
run_no_header --force-merge --db-file "./tests/test-store-dump.1.json" --file "./tests/test-store-dump.2" --output="$tmpdir/tst.merge.out"
diff "./tests/test-store-dump.2.json" "$tmpdir/tst.merge.out"

echo
echo "Checking for differences in output..."
diff -Naur ./tests/test-dump-tool.sh.expected "$tmp" || {
   echo "Found unexpected differences between './tests/test-dump-tool.sh.expected' and '$tmp'"
   exit 1
}

rm -f "$tmp" "$tmpdir"/tst* "$tmpdir"/bad*
rmdir "$tmpdir"
echo "Done"
