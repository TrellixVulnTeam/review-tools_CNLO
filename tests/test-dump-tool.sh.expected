= Test --file =
Running: dump-tool --file ./tests/test-store-dump.1

Running: dump-tool --file ./tests/test-store-dump.2

= Test --file (bad) =
Running: dump-tool --file ./bad --output=stdout
{}

= Test --file --progress =
Running: dump-tool --progress --file ./tests/test-store-dump.1
Parsing snap yamls: ...........
Done: 10 revisions read

= Test --file --warn-if-empty =
Running: dump-tool --warn-if-empty --file ./tests/test-store-dump.1
WARN: Skipping 4ZWSHtNHDBdBsoGCTVgKkk8F9MtvjZS1|1: empty
WARN: Skipping 4ZWSHtNHDBdBsoGCTVgKkk8F9MtvjZS1|2: empty
WARN: Skipping 4ZWSHtNHDBdBsoGCTVgKkk8F9MtvjZS1|3: empty
WARN: Skipping 99T7MUlRhtI3U0QFgl5mXXESAiSwt776|1: empty
WARN: Skipping UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH|1: empty
WARN: Skipping UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH|2: empty
WARN: Skipping UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH|3: empty
WARN: Skipping UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH|4: empty

= Test --db-file =
Running: dump-tool --db-file ./tests/test-store-dump.1.json

Running: dump-tool --db-file ./tests/test-store-dump.2.json

= Test --db-file (bad) =
Running: dump-tool --db-file ./bad --output=stdout
ERROR: Could not read './bad': Expecting value: line 1 column 1 (char 0)

= Test --file --db-file (merge) =
Running: dump-tool --force-merge --db-file ./tests/test-store-dump.1.json --file ./tests/test-store-dump.2

